corehq/motech/dhis2/tasks.py

```diff
@@ -40,15 +39,9 @@ def send_datasets(domain_name, send_now=False, send_date=None):
         if send_now or dataset_map.should_send_on_date(send_date):
             conn = dataset_map.connection_settings
             dataset = dataset_map.get_dataset(send_date)
-            requests = Requests(
-                domain_name,
-                conn.url,
-                conn.username,
-                conn.plaintext_password,
-                verify=not conn.skip_cert_verify,
-                notify_addresses=conn.notify_addresses if hasattr(conn, 'notify_addresses') else None
-            )
-            requests.post('/api/dataValueSets', json=dataset)
+            with conn.get_requests() as requests:
+                requests.post('/api/dataValueSets', json=dataset)
+                conn.update_last_token()
```

Comment:
Should this call to `update_last_token` be the responsibility of the `Requests` object?  Seems like it should happen automatically on `post` if the right auth manager is in use.

Reply:
That sounds right. What about giving the responsibility to AuthManagers that have tokens -- they save the token whenever it is set/refreshed?

Reply:
Like 824a440

Reply:
Yeah, that looks great - I was mostly thinking from an API perspective that you shouldn't need to know about updating tokens in application code when making a request.  That approach looks like it solves that cleanly.